manifests/config/ssl.pp

```diff
@@ -30,31 +32,33 @@
     order   => '02',
   }
 
-  file { $nrpe::nrpe_ssl_dir:
-    ensure => directory,
-    owner  => 'root',
-    group  => $nrpe::nrpe_group,
-    mode   => '0750',
-  }
-  file { "${nrpe::nrpe_ssl_dir}/ca-cert.pem":
-    ensure  => file,
-    owner   => 'root',
-    group   => $nrpe::nrpe_group,
-    mode    => '0640',
-    content => $nrpe::ssl_cacert_file_content,
-  }
-  file { "${nrpe::nrpe_ssl_dir}/nrpe-cert.pem":
-    ensure  => file,
-    owner   => 'root',
-    group   => $nrpe::nrpe_group,
-    mode    => '0640',
-    content => $nrpe::ssl_cert_file_content,
-  }
-  file { "${nrpe::nrpe_ssl_dir}/nrpe-key.pem":
-    ensure  => file,
-    owner   => 'root',
-    group   => $nrpe::nrpe_group,
-    mode    => '0640',
-    content => $nrpe::ssl_privatekey_file_content,
+  if $nrpe::ssl_cert_file_content {
```

Comment:
The condition being on $nrpe::ssl_cert_file_content only looks odd. Regarding TLS in general, I would expect that either CA+Cert+Key are provided or none of them, and anything else is invalid and raise an error.  Some systems might have an optional CA / CRL.  Here the certificate without a key looks odd.

Reply:
I simply move if condition check from parent file to this one.